src/ServerScriptService/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,50 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+local WalkSpeedConfig = require(ReplicatedStorage.Common.Config.WalkSpeed)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local serverComm = Comm.ServerComm.new(character, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._character = character
+	self._comm = serverComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
+
+	self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	character.AncestryChanged:Connect(function()
+		self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	end)
+
+	self._startSprinting = serverComm:WrapMethod(self, "StartSprinting")
+	self._stopSprinting = serverComm:WrapMethod(self, "StopSprinting")
+
+	return self
+end
+
+function Sprint:Destroy()
+	self._trove:Clean()
+end
+
+function Sprint:IsOwner(player: Player?)
+	return player == self._owner
```

Comment:
Since this is server sided, code might get called externally so you should probably do `not player or player == self._owner`, since the server is an owner technically. (I'll eventually break all this ownership stuff out into its own component)

Reply:
@planetme 
You've done `(not not player) or player == self._owner`, but this will just always return true unless a player is passed. The intent is to return `true` when no player is passed for server-authored calls (no player or player is the owner)

Should be `not player or player == self._owner`, or, if it makes a little more sense you can do `player == nil or player == self._owner` which is a little more verbose.